pallets/msa/src/types.rs

```diff
@@ -8,3 +9,10 @@ pub struct AddKeyData {
 	pub msa_id: MessageSenderId,
 	pub nonce: u32,
 }
+
+#[derive(TypeInfo, Debug, Clone, Decode, Encode, PartialEq, Default, MaxEncodedLen)]
+pub struct KeyInfo<BlockNumber> {
+	pub msa_id: MessageSenderId,
+	pub nonce: u32,
```

Comment:
I don't think we need to store every nonce.

Reply:
I was not too crazy about the idea of adding nonce to start with. I thought it was a good idea but should have been left for another story. I do not see any issues with it being here since every key needs to have a nonce. Can you explain why this should go else where?

Reply:
I don't think we even need nonce.

Reply:
I might be missing something but would you expand more on why every key needs to store a nonce? like how that nonce is going to be used later?

Reply:
Keys are used for signing. Any of these keys can sign a transaction.

Reply:
Me and @enddynayn had a conversation about these and the TLDR was that since in capacity the keys are not used in the standard way that the substrate account keys are used, the nonce related logic is not taken care of for any of these keys to prevent **replay attacks** and we need to take care of these ourselves